### ISSUE TITLE
HA discovery switches and flags stopped working

### ISSUE THREAD

**Issue by claesmathias** [ridiekel/jeletask#32]
Using ghcr.io/home-assistant/home-assistant   stable         0b0db9f8d3bc   40 hours ago   1.46GB, auto discovery of switches and flags stopped working.

![Screenshot 2022-11-18 at 13 41 46](https://user-images.githubusercontent.com/1142852/202707422-b3907f4d-e1bc-4af4-b1f5-7f4a721fa644.png)

**Comment by ridiekel** [comment]
I'm going to look into this. I am seeing some changes on the HA config spec. They deprecated some values and removed support for some configs we were using.
I will also start writing some basic tests as jeletask is now used by more than 1 user :-)

**Comment by bertca** [comment]
Hi Ridiekel,
Once again many thanks for the fantastic job, but... ;-)
I am moving my HA from a pi to a nuc and I notice that switches aren't recognised anymore. Do you have any update on this problem?
Thanks.

Bert

**Comment by ridiekel** [comment]
@bertca @claesmathias Could you please test with the latest alpha? `ridiekel/jeletask2mqtt:alpha`

**Comment by ridiekel** [comment]
Also, when you say toggle, you mean the toggle you do using the sliders in the interface like this?

<img width="410" height="96" alt="Image" src="https://github.com/user-attachments/assets/f881cdf5-7d11-4713-b569-cda7ab9c94f4" />

Or is there some other UI element you use?

**Comment by Jieper001** [comment]
Just to chip in, slightly off topic: the reason I don't want to migrate to ProSoft 4.0 is that the (old) IsGUI license afaik does not work with this version anymore.  I have been testing the alpha version with my ProSoft 3.8 and looks good.

**Comment by kristofdegrave** [comment]
Offtopic: Ik heb zelf nu de upgrade naar de 4.0 gedaan. Maar om naar de 5 te gaan, is hiervoor dan een Hardware upgrade nodig?

**Comment by bertca** [comment]
i have 5.0.34 running. When i try 5.0.53 i get : 
Failure
Failed starting container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "java": executable file not found in $PATH: unknown

**Comment by bertca** [comment]
i'm using portainer...

**Comment by bertca** [comment]
step by step...
The problem was caused by the fact that I copied all the ENV variables. I didn't notice there were JAVA variables in between. I removed them and the protainer container is created.
Next problem is to get it starting without error. I attached the logfiles...

[_teletask2mqtt_alpha53_logs.txt](https://github.com/user-attachments/files/23563140/_teletask2mqtt_alpha53_logs.txt)

**Comment by ridiekel** [comment]
Yes, I'm also not sure what is best. That's why I made it configurable. Maybe it's personal taste.

Currently working on a version wher you get more fine grained control over how devices get grouped. https://github.com/ridiekel/jeletask/issues/57#issuecomment-3513167445

Can I close this issue?

**Comment by bertca** [comment]
You can close it. But I think there is some usefull stuff in it for other people who want to upgrade (if they encounter the same problems i had):
- remove TT mqtt device in HA before upgrading
- make sure the ENV variable are set correctly : the RETAIN variable and everything related to JAVA (certainly when you copy ENV in portainer like i did)
- ...

If there are still specific things you want me to chek, please let me know. Thanks for the help (but I'm glad I could point you to a bug too)
